### PR TITLE
Delete mailservers

### LINKED ISSUE #4582: Allow user to delete mailservers
### User Story

As a user
I want to be able to delete mailserver
So I can keep the list manageable

### Description
Part of #4328
The user should be able to delete user-defined mailservers, it should be showing a confirmation dialog before any action is taken.

### PR DESCRIPTION
**Pull request by cammellos**
fixes: #4582 
Allow to delete mailservers

We only allow to delete a mailserver if it's `user-defined` & not currently in use.

### Review notes (optional):
All the code is behind a feature flag, so skip qa.

status: ready

### CODE REVIEW

**Review by janherich — APPROVED** [review]

File: src/status_im/models/mailserver.cljs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+(defn build-url [address password]
+  (let [[initial host] (extract-address-components address)]
+    (str "enode://" initial ":" password "@" host)))
+
+(defn set-input [input-key value {:keys [db]}]
+  {:db (update db :mailservers/manage assoc input-key {:value value
+                                                       :error (case input-key
+                                                                :id   false
+                                                                :name (string/blank? value)
+                                                                :url  (not (utils.inbox/valid-enode-address? value)))})})
```
Comment (rasom) [inline src/status_im/models/mailserver.cljs]: 80 chars

Comment (rasom) [reply src/status_im/models/mailserver.cljs]: oh i see that's not a new code...

Comment (cammellos) [reply src/status_im/models/mailserver.cljs]: that's cool,I will fix that anyway, I introduced it probably before myself.

File: src/status_im/ui/screens/offline_messaging_settings/edit_mailserver/views.cljs [hunk 2]
```diff
@@ -14,15 +15,29 @@
    [status-im.ui.components.text-input.view :as text-input]
    [status-im.ui.screens.offline-messaging-settings.edit-mailserver.styles :as styles]))
 
+(defn handle-delete [id]
+  (utils/show-confirmation (i18n/label :t/delete-mailserver-title)
+                           (i18n/label :t/delete-mailserver-are-you-sure)
+                           (i18n/label :t/delete-mailserver) #(re-frame/dispatch [:delete-mailserver id])))
```
Comment (rasom) [inline src/status_im/ui/screens/offline_messaging_settings/edit_mailserver/views.cljs]: 80 chars

**Review by rasom — APPROVED** [review]

**Review by dmitryn — APPROVED** [review]

**Pull request merged**